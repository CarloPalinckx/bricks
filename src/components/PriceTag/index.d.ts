import { SFC } from 'react';
import { StyledType } from '../../utility/styled';
declare type PartTypeType = 'currency' | 'decimal' | 'fraction' | 'group' | 'infinity' | 'integer' | 'literal' | 'minusSign' | 'nan' | 'plusSign' | 'percentSign';
declare type PartType = {
    type: PartTypeType;
    value: string;
};
declare type PropsType = StyledType & {
    hideCurrency?: boolean;
    superScriptFraction?: boolean;
    showDash?: boolean;
    hideZeros?: boolean;
    parts: Array<PartType>;
    freeLabel?: string;
    displayType?: 'base' | 'action' | 'default';
};
declare const PriceTag: SFC<PropsType>;
export default PriceTag;
export { PropsType, PartTypeType, PartType };
