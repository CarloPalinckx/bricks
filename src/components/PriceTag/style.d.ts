import _R from 'react';
import { StyledComponentClass as _S } from 'styled-components';
import { PropsType } from '.';
import _T from '../../types/ThemeType';
declare type VariantType = {
    color: string;
    fontFamily: string;
    fontSize: string;
    fontWeight: string;
};
declare type PriceTagThemeType = {
    default: VariantType;
    action: VariantType;
    base: VariantType;
};
declare type PriceTagPropsType = {
    displayType?: PropsType['displayType'];
};
declare const StyledPriceTag: _S<PriceTagPropsType & _R.HTMLProps<HTMLSpanElement>, _T, PriceTagPropsType & _R.HTMLProps<HTMLSpanElement>>;
export default StyledPriceTag;
export { PriceTagThemeType, VariantType };
