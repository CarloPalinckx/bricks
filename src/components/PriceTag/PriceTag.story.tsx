import { boolean, number, select, text } from '@storybook/addon-knobs/react';
import { storiesOf } from '@storybook/react';
import React from 'react';
import PriceTag from '.';

storiesOf('PriceTag', module)
    .add('Default', () => (
        <PriceTag
            parts={[
                { type: 'currency' , value: 'â‚¬' },
                { type: 'literal' , value: ' ' },
                { type: 'integer', value: number('integer', 12) },
                { type: 'decimal', value: select(
                    'decimal',
                    [',', '.'],
                    ',',
                )},
                { type: 'fraction', value: select(
                    'fraction',
                    ['00', '34'],
                    '00',
                )},
            ]}
            hideCurrency={boolean('hideCurrency', false)}
            superScriptFraction={boolean('superScriptFraction', false)}
            showDash={boolean('showDash', false)}
            hideZeros={boolean('hideZeros', false)}
            displayType={select(
                'displayType',
                ['action', 'base', 'default'],
                'default',
            )}
            freeLabel={text('freeLabel', 'free!')}
        />
    ));
