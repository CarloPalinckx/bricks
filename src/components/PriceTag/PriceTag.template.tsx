import { StatelessComponent } from 'enzyme';
import React from 'react';
import { StyledType } from '../../utility/styled';
import formatCurrency from './formatters/formatCurrency';
import formatDecimalSeperator from './formatters/formatDecimalSeperator';
import formatFraction from './formatters/formatFraction';

type PartType =
    |'currency'
    |'decimal'
    |'fraction'
    |'group'
    |'infinity'
    |'integer'
    |'literal'
    |'minusSign'
    |'nan'
    |'plusSign'
    |'percentSign';

type PartsType = Array<{
    type:PartType;
    value:string;
}>;

type PropsType = StyledType & {
    hideCurrency?:boolean;
    superScriptFraction?:boolean;
    showDash?:boolean;
    hideZeros?:boolean;
    parts:PartsType;
};

const PriceTag:StatelessComponent<PropsType> = (props):JSX.Element => {
    const isRound = props.parts.reduce(
        (round, part):boolean => {
            return part.type === 'fraction' && parseInt(part.value, 10) === 0
                ? true
                : round;
        },
        false
    );

    const price = props.parts.map((part) => {
        switch (part.type) {
            case 'fraction': return formatFraction(part.value, props, isRound);
            case 'currency': return formatCurrency(part.value, props);
            case 'decimal': return formatDecimalSeperator(part.value, props, isRound);
            default: return part.value;
        }
    });

    return (
        <span className={props.className}>{price}</span>
    );
};

export default PriceTag;
export {
    PropsType,
    PartsType,
    PartType,
};
