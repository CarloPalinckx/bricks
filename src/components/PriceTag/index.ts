export { default as default } from './PriceTag.style';
