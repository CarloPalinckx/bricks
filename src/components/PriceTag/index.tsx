import { StatelessComponent } from 'enzyme';
import React from 'react';
import { StyledType } from '../../utility/styled';
import formatCurrency from './formatters/formatCurrency';
import formatDecimalSeperator from './formatters/formatDecimalSeperator';
import formatFraction from './formatters/formatFraction';
import StyledPriceTag from './style';

type PartTypeType =
    | 'currency'
    | 'decimal'
    | 'fraction'
    | 'group'
    | 'infinity'
    | 'integer'
    | 'literal'
    | 'minusSign'
    | 'nan'
    | 'plusSign'
    | 'percentSign';

type PartType = {
    type: PartTypeType;
    value: string;
};

type PropsType = StyledType & {
    hideCurrency?: boolean;
    superScriptFraction?: boolean;
    showDash?: boolean;
    hideZeros?: boolean;
    parts: Array<PartType>;
    freeLabel?: string;
    displayType?: 'base' | 'action' | 'default';
};

type StatsType = {
    isRound: boolean;
    isFree: boolean;
};

const isFree = (part: PartType): boolean =>
    (part.type === 'integer' || part.type === 'fraction') && parseInt(part.value, 10) !== 0;

const isRound = (part: PartType): boolean => part.type === 'fraction' && parseInt(part.value, 10) === 0;

const deriveStatsFromPart = (initialStats: StatsType, part: PartType): StatsType => ({
    isRound: isRound(part) ? true : initialStats.isRound,
    isFree: isFree(part) ? false : initialStats.isFree,
});

const PriceTag: StatelessComponent<PropsType> = (props): JSX.Element => {
    const stats = props.parts.reduce(deriveStatsFromPart, {
        isRound: false,
        isFree: true,
    });

    const price = props.parts.map(part => {
        switch (part.type) {
            case 'fraction':
                return formatFraction(part.value, props, stats.isRound);
            case 'currency':
                return formatCurrency(part.value, props);
            case 'decimal':
                return formatDecimalSeperator(part.value, props, stats.isRound);
            default:
                return part.value;
        }
    });

    return (
        <StyledPriceTag displayType={props.displayType}>
            {stats.isFree && props.freeLabel !== undefined ? props.freeLabel : price}
        </StyledPriceTag>
    );
};

export default PriceTag;
export { PropsType, PartTypeType, PartType };
