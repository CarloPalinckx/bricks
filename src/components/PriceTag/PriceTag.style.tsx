import { StyledComponentClass as _S } from 'styled-components';
import styled from '../../utility/styled';
import PriceTag from './PriceTag.template';

const StyledPriceTag = styled(PriceTag)`
    color: ${({ theme }):string => theme.priceTag.color};
    font-family: ${({ theme }):string => theme.priceTag.fontFamily};
    font-size: ${({ theme }):string => theme.priceTag.fontSize};
    font-weight: ${({ theme }):string => theme.priceTag.fontWeight};

    sup {
        font-size: ${({ theme }):string => theme.priceTag.sup.fontSize};
        font-weight: ${({ theme }):string => theme.priceTag.sup.fontWeight};
    }
`;

export default StyledPriceTag;
