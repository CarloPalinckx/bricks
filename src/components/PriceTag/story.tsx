import { boolean, number, select, text } from '@storybook/addon-knobs/react';
import { storiesOf } from '@storybook/react';
import React from 'react';
import PriceTag, { PropsType } from '.';

storiesOf('PriceTag', module).add('Default', () => (
    <PriceTag
        parts={
            [
                { type: 'currency', value: 'â‚¬' },
                { type: 'literal', value: ' ' },
                { type: 'integer', value: number('integer', 12) },
                {
                    type: 'decimal',
                    value: select('decimal', [',', '.'], ','),
                },
                {
                    type: 'fraction',
                    value: select('fraction', ['00', '34'], '00'),
                },
            ] as PropsType['parts']
        }
        hideCurrency={boolean('hideCurrency', false)}
        superScriptFraction={boolean('superScriptFraction', false)}
        showDash={boolean('showDash', false)}
        hideZeros={boolean('hideZeros', false)}
        displayType={select('displayType', ['action', 'base', 'default'], 'default') as PropsType['displayType']}
        freeLabel={text('freeLabel', 'free!')}
    />
));
