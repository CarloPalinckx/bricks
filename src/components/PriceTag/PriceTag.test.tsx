import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import PriceTag from '.';
import MosTheme from '../../themes/MosTheme';

describe('PriceTag',  () => {
    it('should render with a hidden currency', () => {
        const component = mount(
            <MosTheme>
                <PriceTag
                    parts={[
                        { type: 'currency' , value: '€' },
                        { type: 'literal' , value: ' ' },
                        { type: 'integer', value: '10' },
                        { type: 'decimal', value: ',' },
                        { type: 'fraction', value: '20' },
                    ]}
                    hideCurrency
                />
            </MosTheme>
        ).find(PriceTag);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render with a hidden currency and superscript franction', () => {
        const component = mount(
            <MosTheme>
                <PriceTag
                    parts={[
                        { type: 'currency' , value: '€' },
                        { type: 'literal' , value: ' ' },
                        { type: 'integer', value: '10' },
                        { type: 'decimal', value: ',' },
                        { type: 'fraction', value: '20' },
                    ]}
                    superScriptFraction
                    hideCurrency
                />
            </MosTheme>
        ).find(PriceTag);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render with a dashed fraction', () => {
        const component = mount(
            <MosTheme>
                <PriceTag
                    parts={[
                        { type: 'currency' , value: '€' },
                        { type: 'literal' , value: ' ' },
                        { type: 'integer', value: '10' },
                        { type: 'decimal', value: ',' },
                        { type: 'fraction', value: '00' },
                    ]}
                    showDash
                />
            </MosTheme>
        ).find(PriceTag);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render with a hidden fraction on a round value', () => {
        const component = mount(
            <MosTheme>
                <PriceTag
                    parts={[
                        { type: 'currency' , value: '€' },
                        { type: 'literal' , value: ' ' },
                        { type: 'integer', value: '10' },
                        { type: 'decimal', value: ',' },
                        { type: 'fraction', value: '00' },
                    ]}
                    hideZeros
                />
            </MosTheme>
        ).find(PriceTag);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render with a fraction in superscript', () => {
        const component = mount(
            <MosTheme>
                <PriceTag
                    parts={[
                        { type: 'currency' , value: '€' },
                        { type: 'literal' , value: ' ' },
                        { type: 'integer', value: '10' },
                        { type: 'decimal', value: ',' },
                        { type: 'fraction', value: '20' },
                    ]}
                    superScriptFraction
                />
            </MosTheme>
        ).find(PriceTag);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render with showDash and superscript on a round value', () => {
        const component = mount(
            <MosTheme>
                <PriceTag
                    parts={[
                        { type: 'currency' , value: '€' },
                        { type: 'literal' , value: ' ' },
                        { type: 'integer', value: '10' },
                        { type: 'decimal', value: ',' },
                        { type: 'fraction', value: '00' },
                    ]}
                    showDash
                    superScriptFraction
                />
            </MosTheme>
        ).find(PriceTag);

        expect(toJson(component)).toMatchSnapshot();
    });
});
