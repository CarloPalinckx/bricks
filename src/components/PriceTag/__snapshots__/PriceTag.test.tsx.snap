// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceTag should render a base price 1`] = `
.c0 {
  position: relative;
  color: #88979d;
  font-family: Source Sans Pro,sans-serif;
  font-size: 22px;
  font-weight: 400;
}

.c0 sup {
  font-size: .7em;
}

.c0::after {
  content: '';
  width: 100%;
  height: 2px;
  left: 0;
  top: 50%;
  margin-top: -2px;
  position: absolute;
  background: #88979d;
  opacity: .7;
  -webkit-transform: rotate(-8deg);
  -ms-transform: rotate(-8deg);
  transform: rotate(-8deg);
}

<Styled(PriceTag)
  displayType="base"
  parts={
    Array [
      Object {
        "type": "currency",
        "value": "€",
      },
      Object {
        "type": "literal",
        "value": " ",
      },
      Object {
        "type": "integer",
        "value": "10",
      },
      Object {
        "type": "decimal",
        "value": ",",
      },
      Object {
        "type": "fraction",
        "value": "20",
      },
    ]
  }
>
  <PriceTag
    className="c0"
    displayType="base"
    parts={
      Array [
        Object {
          "type": "currency",
          "value": "€",
        },
        Object {
          "type": "literal",
          "value": " ",
        },
        Object {
          "type": "integer",
          "value": "10",
        },
        Object {
          "type": "decimal",
          "value": ",",
        },
        Object {
          "type": "fraction",
          "value": "20",
        },
      ]
    }
  >
    <span
      className="c0"
    >
      €
       
      10
      ,
      20
    </span>
  </PriceTag>
</Styled(PriceTag)>
`;

exports[`PriceTag should render a label with the price is 0 1`] = `
.c0 {
  position: relative;
  color: #333740;
  font-family: Source Sans Pro,sans-serif;
  font-size: 28px;
  font-weight: 400;
}

.c0 sup {
  font-size: .7em;
}

<Styled(PriceTag)
  freeLabel="free"
  parts={
    Array [
      Object {
        "type": "currency",
        "value": "€",
      },
      Object {
        "type": "literal",
        "value": " ",
      },
      Object {
        "type": "integer",
        "value": "0",
      },
      Object {
        "type": "decimal",
        "value": ",",
      },
      Object {
        "type": "fraction",
        "value": "00",
      },
    ]
  }
>
  <PriceTag
    className="c0"
    freeLabel="free"
    parts={
      Array [
        Object {
          "type": "currency",
          "value": "€",
        },
        Object {
          "type": "literal",
          "value": " ",
        },
        Object {
          "type": "integer",
          "value": "0",
        },
        Object {
          "type": "decimal",
          "value": ",",
        },
        Object {
          "type": "fraction",
          "value": "00",
        },
      ]
    }
  >
    <span
      className="c0"
    >
      free
    </span>
  </PriceTag>
</Styled(PriceTag)>
`;

exports[`PriceTag should render an action price 1`] = `
.c0 {
  position: relative;
  color: #6bde78;
  font-family: Source Sans Pro,sans-serif;
  font-size: 28px;
  font-weight: 400;
}

.c0 sup {
  font-size: .7em;
}

<Styled(PriceTag)
  displayType="action"
  parts={
    Array [
      Object {
        "type": "currency",
        "value": "€",
      },
      Object {
        "type": "literal",
        "value": " ",
      },
      Object {
        "type": "integer",
        "value": "10",
      },
      Object {
        "type": "decimal",
        "value": ",",
      },
      Object {
        "type": "fraction",
        "value": "20",
      },
    ]
  }
>
  <PriceTag
    className="c0"
    displayType="action"
    parts={
      Array [
        Object {
          "type": "currency",
          "value": "€",
        },
        Object {
          "type": "literal",
          "value": " ",
        },
        Object {
          "type": "integer",
          "value": "10",
        },
        Object {
          "type": "decimal",
          "value": ",",
        },
        Object {
          "type": "fraction",
          "value": "20",
        },
      ]
    }
  >
    <span
      className="c0"
    >
      €
       
      10
      ,
      20
    </span>
  </PriceTag>
</Styled(PriceTag)>
`;

exports[`PriceTag should render with a dashed fraction 1`] = `
.c0 {
  position: relative;
  color: #333740;
  font-family: Source Sans Pro,sans-serif;
  font-size: 28px;
  font-weight: 400;
}

.c0 sup {
  font-size: .7em;
}

<Styled(PriceTag)
  parts={
    Array [
      Object {
        "type": "currency",
        "value": "€",
      },
      Object {
        "type": "literal",
        "value": " ",
      },
      Object {
        "type": "integer",
        "value": "10",
      },
      Object {
        "type": "decimal",
        "value": ",",
      },
      Object {
        "type": "fraction",
        "value": "00",
      },
    ]
  }
  showDash={true}
>
  <PriceTag
    className="c0"
    parts={
      Array [
        Object {
          "type": "currency",
          "value": "€",
        },
        Object {
          "type": "literal",
          "value": " ",
        },
        Object {
          "type": "integer",
          "value": "10",
        },
        Object {
          "type": "decimal",
          "value": ",",
        },
        Object {
          "type": "fraction",
          "value": "00",
        },
      ]
    }
    showDash={true}
  >
    <span
      className="c0"
    >
      €
       
      10
      ,
      -
    </span>
  </PriceTag>
</Styled(PriceTag)>
`;

exports[`PriceTag should render with a fraction in superscript 1`] = `
.c0 {
  position: relative;
  color: #333740;
  font-family: Source Sans Pro,sans-serif;
  font-size: 28px;
  font-weight: 400;
}

.c0 sup {
  font-size: .7em;
}

<Styled(PriceTag)
  parts={
    Array [
      Object {
        "type": "currency",
        "value": "€",
      },
      Object {
        "type": "literal",
        "value": " ",
      },
      Object {
        "type": "integer",
        "value": "10",
      },
      Object {
        "type": "decimal",
        "value": ",",
      },
      Object {
        "type": "fraction",
        "value": "20",
      },
    ]
  }
  superScriptFraction={true}
>
  <PriceTag
    className="c0"
    parts={
      Array [
        Object {
          "type": "currency",
          "value": "€",
        },
        Object {
          "type": "literal",
          "value": " ",
        },
        Object {
          "type": "integer",
          "value": "10",
        },
        Object {
          "type": "decimal",
          "value": ",",
        },
        Object {
          "type": "fraction",
          "value": "20",
        },
      ]
    }
    superScriptFraction={true}
  >
    <span
      className="c0"
    >
      €
       
      10
      ,
      <sup
        key="fraction"
      >
        20
      </sup>
    </span>
  </PriceTag>
</Styled(PriceTag)>
`;

exports[`PriceTag should render with a hidden currency 1`] = `
.c0 {
  position: relative;
  color: #333740;
  font-family: Source Sans Pro,sans-serif;
  font-size: 28px;
  font-weight: 400;
}

.c0 sup {
  font-size: .7em;
}

<Styled(PriceTag)
  hideCurrency={true}
  parts={
    Array [
      Object {
        "type": "currency",
        "value": "€",
      },
      Object {
        "type": "literal",
        "value": " ",
      },
      Object {
        "type": "integer",
        "value": "10",
      },
      Object {
        "type": "decimal",
        "value": ",",
      },
      Object {
        "type": "fraction",
        "value": "20",
      },
    ]
  }
>
  <PriceTag
    className="c0"
    hideCurrency={true}
    parts={
      Array [
        Object {
          "type": "currency",
          "value": "€",
        },
        Object {
          "type": "literal",
          "value": " ",
        },
        Object {
          "type": "integer",
          "value": "10",
        },
        Object {
          "type": "decimal",
          "value": ",",
        },
        Object {
          "type": "fraction",
          "value": "20",
        },
      ]
    }
  >
    <span
      className="c0"
    >
       
      10
      ,
      20
    </span>
  </PriceTag>
</Styled(PriceTag)>
`;

exports[`PriceTag should render with a hidden currency and superscript franction 1`] = `
.c0 {
  position: relative;
  color: #333740;
  font-family: Source Sans Pro,sans-serif;
  font-size: 28px;
  font-weight: 400;
}

.c0 sup {
  font-size: .7em;
}

<Styled(PriceTag)
  hideCurrency={true}
  parts={
    Array [
      Object {
        "type": "currency",
        "value": "€",
      },
      Object {
        "type": "literal",
        "value": " ",
      },
      Object {
        "type": "integer",
        "value": "10",
      },
      Object {
        "type": "decimal",
        "value": ",",
      },
      Object {
        "type": "fraction",
        "value": "20",
      },
    ]
  }
  superScriptFraction={true}
>
  <PriceTag
    className="c0"
    hideCurrency={true}
    parts={
      Array [
        Object {
          "type": "currency",
          "value": "€",
        },
        Object {
          "type": "literal",
          "value": " ",
        },
        Object {
          "type": "integer",
          "value": "10",
        },
        Object {
          "type": "decimal",
          "value": ",",
        },
        Object {
          "type": "fraction",
          "value": "20",
        },
      ]
    }
    superScriptFraction={true}
  >
    <span
      className="c0"
    >
       
      10
      ,
      <sup
        key="fraction"
      >
        20
      </sup>
    </span>
  </PriceTag>
</Styled(PriceTag)>
`;

exports[`PriceTag should render with a hidden fraction on a round value 1`] = `
.c0 {
  position: relative;
  color: #333740;
  font-family: Source Sans Pro,sans-serif;
  font-size: 28px;
  font-weight: 400;
}

.c0 sup {
  font-size: .7em;
}

<Styled(PriceTag)
  hideZeros={true}
  parts={
    Array [
      Object {
        "type": "currency",
        "value": "€",
      },
      Object {
        "type": "literal",
        "value": " ",
      },
      Object {
        "type": "integer",
        "value": "10",
      },
      Object {
        "type": "decimal",
        "value": ",",
      },
      Object {
        "type": "fraction",
        "value": "00",
      },
    ]
  }
>
  <PriceTag
    className="c0"
    hideZeros={true}
    parts={
      Array [
        Object {
          "type": "currency",
          "value": "€",
        },
        Object {
          "type": "literal",
          "value": " ",
        },
        Object {
          "type": "integer",
          "value": "10",
        },
        Object {
          "type": "decimal",
          "value": ",",
        },
        Object {
          "type": "fraction",
          "value": "00",
        },
      ]
    }
  >
    <span
      className="c0"
    >
      €
       
      10
    </span>
  </PriceTag>
</Styled(PriceTag)>
`;

exports[`PriceTag should render with showDash and superscript on a round value 1`] = `
.c0 {
  position: relative;
  color: #333740;
  font-family: Source Sans Pro,sans-serif;
  font-size: 28px;
  font-weight: 400;
}

.c0 sup {
  font-size: .7em;
}

<Styled(PriceTag)
  parts={
    Array [
      Object {
        "type": "currency",
        "value": "€",
      },
      Object {
        "type": "literal",
        "value": " ",
      },
      Object {
        "type": "integer",
        "value": "10",
      },
      Object {
        "type": "decimal",
        "value": ",",
      },
      Object {
        "type": "fraction",
        "value": "00",
      },
    ]
  }
  showDash={true}
  superScriptFraction={true}
>
  <PriceTag
    className="c0"
    parts={
      Array [
        Object {
          "type": "currency",
          "value": "€",
        },
        Object {
          "type": "literal",
          "value": " ",
        },
        Object {
          "type": "integer",
          "value": "10",
        },
        Object {
          "type": "decimal",
          "value": ",",
        },
        Object {
          "type": "fraction",
          "value": "00",
        },
      ]
    }
    showDash={true}
    superScriptFraction={true}
  >
    <span
      className="c0"
    >
      €
       
      10
      ,
      -
    </span>
  </PriceTag>
</Styled(PriceTag)>
`;
